### ISSUE TITLE
nightly-20240827 perf degradation

### ISSUE THREAD

**Issue by cyliu0** [risingwavelabs/risingwave#18281]
### Describe the bug

```+---------------------------------------------------------------+--------------+------------+----------------------------------------------------+---------------------+-----------------------------+-------------------------------+
| BENCHMARK NAME                                                | EXECUTION ID | STATUS     | KEY METRICS                                        | FLUCTUATION OF BEST | FLUCTUATION OF LAST 10 DAYS | FLUCTUATION OF LAST EXECUTION |
+---------------------------------------------------------------+--------------+------------+----------------------------------------------------+---------------------+-----------------------------+-------------------------------+
| nexmark-q0-blackhole-medium-1cn                               |        37137 | Negative   | avg-source-output-rows-per-second                  | -57.36%             | -41.78%                     | -45.93%                       |
| nexmark-q7-blackhole-medium-1cn                               |        37162 | Negative   | avg-source-output-rows-per-second                  | -23.03%             | -12.47%                     | -17.49%                       |
| nexmark-q7-rewrite-blackhole-medium-1cn                       |        37166 | Negative   | avg-source-output-rows-per-second                  | -54.93%             | -38.21%                     | -42.59%                       |
| nexmark-q8-blackhole-medium-1cn                               |        37168 | Negative   | avg-source-output-rows-per-second                  | -37.41%             | -10.71%                     | -14.31%                       |
| nexmark-q12-blackhole-medium-1cn                              |        37174 | Negative   | avg-source-output-rows-per-second                  | -41.03%             | -30.40%                     | -34.68%                       |
| nexmark-q13-blackhole-medium-1cn                              |        37180 | Negative   | avg-source-output-rows-per-second                  | -47.55%             | -35.62%                     | -39.14%                       |
| nexmark-q3-no-condition-blackhole-medium-1cn                  |        37184 | Negative   | avg-source-output-rows-per-second                  | -38.30%             | -19.83%                     | -24.15%                       |
```

### The version of RisingWave

nightly-20240827

### Additional context

https://github.com/risingwavelabs/rw-commits-history?tab=readme-ov-file#nightly-20240827

**Comment by lmatz** [comment]
Found by @Li0k 
![image (4)](https://github.com/user-attachments/assets/1bb5bc8c-2a4f-48e0-8763-fc351fc621ee)
![image (5)](https://github.com/user-attachments/assets/9a6699b4-2fda-4a67-b8d5-913a326c7592)
that the second half period has 0 source throughput

same root cause as https://github.com/risingwavelabs/risingwave/issues/18268#issuecomment-2312598776

**Comment by lmatz** [comment]
solved by #18302
